### ISSUE TITLE
Automatically sort query/mutations for introspection and in exported GraphQL schema

### ISSUE THREAD

**Issue by smhutch** [strawberry-graphql/strawberry#3149]
## Feature Request Type

- [ ] Core functionality
- [x] Alteration (enhancement/optimization) of existing feature(s)
- [ ] New behavior

## Description

By default, the GraphQL schema stawberry generates uses the same ordering as the order fields in which fields are defined. This makes the `/graphql` introspection UI hard to parse. As a consumer of this UI and the exported schema.graphql files, it's very hard to find all query fields associated with some concept (like `userBy...`) because these are not logically grouped together.

It would be amazing if stawberry could (either by default or as a configuration option) support running `lexicographicSortSchema` ([reference](https://graphql-js.org/api/function/lexicographicsortschema/)) on the schema. This would take care of updating both the introspection UI, and exported schema file.

### Current output
![CleanShot 2023-10-13 at 10 03 10@2x](https://github.com/strawberry-graphql/strawberry/assets/7217244/f7ce8759-b48d-43da-9deb-0a14a9685162)

### Desired output
![CleanShot 2023-10-13 at 10 06 25@2x](https://github.com/strawberry-graphql/strawberry/assets/7217244/a008d886-c1ac-411b-9fe6-5369e7b95e92)

**Comment by DoctorJohn** [comment]
We could make this an option. Shouldn't be too hard either since we have access to the "lexicographic sort schema" function [via graphql-core](https://graphql-core-3.readthedocs.io/en/latest/modules/utilities.html#graphql.utilities.lexicographic_sort_schema).

**Comment by Nalin7parihar** [comment]
Hi! I'd like to work on this issue for Hacktoberfest. Could you please confirm if this issue is still relevant?

**Comment by DoctorJohn** [comment]
> Hi! I'd like to work on this issue for Hacktoberfest. Could you please confirm if this issue is still relevant?

It's still relevant, feel free to work on it!
